tests/v2/validation/projects/projects_additional_test.go

```diff
@@ -0,0 +1,150 @@
+	finalizer := append([]string{dummyFinalizer}, createdProject.Finalizers...)
+	_, err = updateProjectNamespaceFinalizer(pr.client, createdProject.Name, createdProject.Namespace, finalizer)
+	require.NoError(pr.T(), err, "Failed to update finalizer.")
+
+	log.Info("Delete the Project.")
+	err = projectsApi.DeleteProject(pr.client, createdProject)
+	require.NoError(pr.T(), err)
+
+	log.Info("Add the standard user to the downstream cluster as cluster owner.")
+	err = users.AddClusterRoleToUser(pr.client, pr.cluster, createdUser, roleOwner, nil)
+	require.NoError(pr.T(), err)
+
+	testEndTime := time.Now()
+
+	log.Info("Verify that there are no errors in the Rancher logs related to role binding.")
+	leaderPodName, err := getRancherLeaderPod(pr.client)
+	require.NoError(pr.T(), err)
+	errorRegex := `\[ERROR\] error syncing '(.*?)': handler mgmt-auth-crtb-controller: .*? namespaces "(.*?)" (?:not found|is forbidden), requeuing`
+	err = checkPodLogsForErrors(pr.client, localCluster, leaderPodName, RancherNamespace, errorRegex, testStartTime, testEndTime)
+	require.NoError(pr.T(), err)
+
+	log.Info("Remove the finalizer that was previously added to the project.")
+	finalizer = nil
+	_, err = updateProjectNamespaceFinalizer(pr.client, createdProject.Name, createdProject.Namespace, finalizer)
+	require.NoError(pr.T(), err, "Failed to update finalizer.")
+
+	log.Info("Verify that the project is deleted.")
+	listOpt := metav1.ListOptions{
+		FieldSelector: "metadata.name=" + createdProject.Name,
+	}
+	project, err := projectsApi.ListProjects(pr.client, createdProject.Namespace, listOpt)
+	require.NoError(pr.T(), err)
+	require.Empty(pr.T(), project, "Project still exists.")
+}
+
+func (pr *ProjectsAdditionalTestSuite) TestUserAdditionToProjectWithTerminatingProjectNamespace() {
+	subSession := pr.session.NewSession()
+	defer subSession.Cleanup()
+
+	log.Info("Create a standard user.")
+	createdUser, err := users.CreateUserWithRole(pr.client, users.UserConfig(), standardUser)
+	require.NoError(pr.T(), err)
+	pr.T().Logf("Created user: %v", createdUser.Username)
+
+	log.Info("Create a project in the downstream cluster.")
+	createdProject, err := createProject(pr.client, pr.cluster.ID)
+	require.NoError(pr.T(), err)
+	err = waitForFinalizerToUpdate(pr.client, createdProject.Name)
+	require.NoError(pr.T(), err)
+
+	log.Info("Simulate a project stuck in terminating state by adding a finalizer to the project.")
+	finalizer := append([]string{dummyFinalizer}, createdProject.Finalizers...)
+	_, err = updateProjectNamespaceFinalizer(pr.client, createdProject.Name, createdProject.Namespace, finalizer)
+	require.NoError(pr.T(), err, "Failed to update finalizer.")
+
+	log.Info("Delete the Project.")
+	err = projectsApi.DeleteProject(pr.client, createdProject)
+	require.NoError(pr.T(), err)
+
+	log.Info("Add the standard user to the project as project owner.")
+	_, err = createProjectRoleTemplateBinding(pr.client, createdUser, createdProject, roleProjectOwner)
```

Comment:
Since the plan is to use public APIs that are available now for any new tests for Projects, I used createProjectRoleTemplateBinding that is using public API under the hood. Isn't AddProjectMember using Norman?

Reply:
Since we are creating projects using the public API, the AddProjectMember won't accept the project we created as valid input because it is not a Norman resource, right?